src/fastmath.rs

```diff
@@ -1,3 +1,7 @@
+#![allow(clippy::cast_possible_wrap)]
+#![allow(clippy::cast_precision_loss)]
+#![allow(clippy::cast_sign_loss)]
```

Comment:
Why do these need to be added?

Reply:
`powf` has some code that causes clippy warnings. For example `x.to_bits() as i32` can wrap. If you want, i can go further into it, but IMO the warnings indicate problems that either can't happen with our assumptions or don't matter (wrapping doesn't matter because we're just using the value for bitwise operations).